### PR TITLE
Add tests for the plugin

### LINKED ISSUE #2: Add self-CI
We should follow the example set by [`julia-buildkite-plugin` and set up self-CI](https://github.com/JuliaCI/julia-buildkite-plugin/blob/ced91aedc51b4e28830a59f8cfc40fd5dcfc2346/.buildkite/run_tests.yml).

We should run for the following configurations at first:

* x86_64-linux-gnu
* x86_64-apple-darwin
* aarch64-apple-darwin
* x86_64-w64-mingw32

It should be fairly straightforward to setup a templating system; unfortunately buildkite's `matrix` setup doesn't work with plugin attributes (😱), so we will need to instead do something like what the [`julia-buildkite-plugin` does and manually export bash variables](https://github.com/JuliaCI/julia-buildkite-plugin/blob/ced91aedc51b4e28830a59f8cfc40fd5dcfc2346/.buildkite/pipeline.yml#L6-L27) that are then used in our workhorse test runner.

One strange detail about self-testing is that, in order to properly use the current repository as a plugin within a pipeline, we have to install it into the "vendored plugin" location, which means that we add [these steps](https://github.com/JuliaCI/julia-buildkite-plugin/blob/ced91aedc51b4e28830a59f8cfc40fd5dcfc2346/.buildkite/run_tests.yml#L1-L4) as a [`metahook` plugin invocation](https://github.com/JuliaCI/julia-buildkite-plugin/blob/ced91aedc51b4e28830a59f8cfc40fd5dcfc2346/.buildkite/run_tests.yml#L13-L14) to all of our steps.  That allows us to [then use a relative path as a plugin source](https://github.com/JuliaCI/julia-buildkite-plugin/blob/ced91aedc51b4e28830a59f8cfc40fd5dcfc2346/.buildkite/run_tests.yml#L15-L16).

### PR DESCRIPTION
**Pull request by thazhemadam**
Add tests for the plugin that ensures that a specified version of `ytt` gets installed properly and is usable in the expected manner.

### CODE REVIEW

**Comment by staticfloat** [comment]
I think in the buildkite webui, the agents declaration may be top-level. That’s probably what is causing the windows agents or search for a “sandbox.jl true” agent, which doesn’t exist. ;)

**Comment by thazhemadam** [comment]
> I think in the buildkite webui, the agents declaration may be top-level

Indentation struggles from editing on mobile devices strikes again! Fixed.
Windows seems to be failing at the download step. Marking this draft so I can debug that a bit more and figure out what's happening there (likely some path-related shenanigans).

File: hooks/pre-command [hunk 1]
```diff
@@ -61,8 +61,10 @@ if [[ ! -d "${DOWNLOAD_DESTINATION}" ]]; then
 fi
 
 URL="https://github.com/vmware-tanzu/carvel-ytt/releases/download/v${YTT_VERSION}/ytt-${OS}-${ARCH}"
+echo "Downloading $URL..."
```
Comment (staticfloat) [inline hooks/pre-command]: You need a `.exe` at the end of this for windows.  :)

Comment (thazhemadam) [reply hooks/pre-command]: Fixed in b1da3dc!

**Comment by staticfloat** [comment]
Great!  Next step is to use it to actually do some templating, within the test suite here.  :)

**Pull request merged**